math/mathcore/src/GoFTest.cxx

```diff
@@ -930,11 +929,11 @@ void GoFTest::AndersonDarling2SamplesTest(Double_t& pvalue, Double_t& testStat)
          return;
       }
       Double_t Fo = 0.0, Dn = 0.0;
-      UInt_t n = fSamples[0].size();
-      for (UInt_t i = 0; i < n; ++i) {
+      size_t n = fSamples[0].size();
+      for (size_t i = 0; i < n; ++i) {
          Double_t Fn = (i + 1.0) / n;
          Double_t F = (*fCDF)(fSamples[0][i]);
-         Double_t result = std::max(TMath::Abs(Fn - F), TMath::Abs(Fo - Fn));
+         Double_t result = std::max(TMath::Abs(Fn - F), TMath::Abs(Fo - F));
```

Comment:
```suggestion
         double result = std::abs(Fn - F);
```
Why is this `std::max` comparison with the distance for the previous point necessary? If I understand the logic of the algorithm correctly, you want to find the maximum of `abs(Fn - F)` for all `i`.

Reply:
I think the max should stay, because you wanter each data point i  the maximum between the difference of the theoretical CDF function value, F(x(i)) with the empirical CDF value at x(i), Fn  and  the empirical CDF value at the point before Fo. The bug, described in #9636 was exactly here, because the maximum was computed using the difference Fn with Fo, which was wrong instead of F with Fo.

Reply:
Ok, thanks for clarifying!